src/notation/internal/notationparts.cpp

```diff
@@ -481,6 +481,8 @@ void NotationParts::appendStaff(Staff* staff, const ID& destinationPartId)
 
     Part* destinationPart = partModifiable(destinationPartId);
     if (!staff || !destinationPart) {
+        staff->unlink();
```

Comment:
`staff` may be null at this point, in which case the added lines will cause a crash. Better create a separate assert for `staff`:
```
    TRACEFUNC;

    IF_ASSERT_FAILED(staff) {
        return;
    }

    Part* destinationPart = partModifiable(destinationPartId);
    if (!destinationPart) {
        staff->unlink();
        ...
```
(I think it's fair to use an assert for `staff`, isn't it?)

(same in `appendLinkedStaff`)

Reply:
Oh, really good catch, I dont know where my brain is today

Reply:
Is there a reason that's a better choice than 

`if (!staff || !destinationPart) {
        if (staff) {
            staff->unlink();
            delete staff;
        }
        return;
    }`

Just a regular null check?

Reply:
Mainly for semantical reasons, I would use an assert for `staff`, to indicate that we consider it a mistake if `staff` is null. But functionally it won't really make a difference. 

(In any case I personally think it would look clearer if we'd do the check for `staff` separately, at the top of the function, instead of inside the `if (!staff || !destinationPart)` block.)